app/models/user.rb

```diff
@@ -1,3 +1,9 @@
 class User < ApplicationRecord
   authenticates_with_sorcery!
+
+  validates :password, length: {minimum: 3}, if: -> {:new_record? || changes[:crypted_password]}
+  validates :password, confirmation: true, if: -> {:new_record? || changes[:crypted_password]}
+  validates :password_confirmation, presence: true, if: -> {:new_record? || changes || [:crypted_password]}
```

Comment:
this looks like a typo
```suggestion
  validates :password_confirmation, presence: true, if: -> {:new_record? || changes[:crypted_password]}
```